tests/filesystem_tests/test_hdfs_handler.py

```diff
@@ -37,6 +37,58 @@ def setUp(self):
         self.fs = "hdfs"
         self.tmpfile_name = "tmpfile.txt"
 
+    def test_root(self):
+        with chainerio.create_handler(self.fs) as handler:
+            # prepare the tmpfile
+            tmpdir = "testdir"
+
+            handler.mkdir(tmpdir)
+            try:
+                self.assertTrue(handler.exists(tmpdir))
+
+                testfile_name = "testfile"
+                testfile_path = os.path.join(tmpdir, testfile_name)
+                with handler.open(testfile_path, "w") as f:
+                    f.write(self.test_string)
+
+                self.assertTrue(handler.exists(testfile_path))
+
+                # set root
+                handler.root = os.path.join(tmpdir)
+                # open with root
+                with handler.open(testfile_name, "r") as f:
+                    self.assertEqual(self.test_string, f.read())
+
+                # list with root
+                self.assertEqual(list(handler.list()), [testfile_name])
+                # stat with root
+                self.assertIsNotNone(handler.stat(testfile_name))
+                # exits with root
+                self.assertTrue(handler.exists(testfile_name))
+                # exits with root
+                new_testfile_name = "new_test_filename"
+                handler.rename(testfile_name, new_testfile_name)
+                self.assertFalse(handler.exists(testfile_name))
+                self.assertTrue(handler.exists(new_testfile_name))
+                # mkdir with root
+                mkdir_name = "mkdir"
+                handler.mkdir(mkdir_name)
+                self.assertTrue(handler.exists(mkdir_name))
+                # # mkdir with root
+                # TODO(tianqi): has a bug with makedirs,
+                # will fix it in another PR
+                # mkdirs_name = "mkdirs/dirs"
+                # handler.makedirs(mkdirs_name)
+                # self.assertTrue(handler.exists(mkdirs_name))
+                # isdir with root
+                self.assertTrue(handler.isdir(mkdir_name))
+                # remove with root
+                handler.remove(mkdir_name)
+                self.assertFalse(handler.exists(mkdir_name))
+            finally:
+                handler.root = ""
```

Comment:
The `.root` is set back to delete the `testdir` as since the `.root` was set to `testdir` at https://github.com/chainer/chainerio/blob/dea9e5a727d87caf94d7341264bd7cec41e193dd/tests/filesystem_tests/test_hdfs_handler.py#L57

Hence we need to move to the default root before deleting the `testdir`. I added a comment to improve the readability